src/OrchardCore.Modules/OrchardCore.Users/Views/TwoFactorAuthentication/Index.cshtml

```diff
@@ -1,82 +1,180 @@
+                        }
+                    </select>
+                </div>
 
-            <a asp-action="GenerateRecoveryCodes" class="btn btn-primary">@T["Reset recovery codes"]</a>
+                <button type="submit" class="btn btn-sm btn-primary">@T["Update"]</button>
+            </form>
         }
     </div>
 
-    <div class="text-center">
-        <h3 class="text-center">@T["Authenticator App"]</h3>
-        @if (!Model.HasAuthenticator)
-        {
-            <a id="enable-authenticator" asp-action="EnableAuthenticator" class="btn btn-primary">@T["Add authenticator app"]</a>
-        }
-        else
+    <div class="card-header">
+        @T["Two-factor methods"]
+    </div>
+
+    <div class="card-body p-0">
+        @foreach (var methodShape in Model.AuthenticationMethods)
         {
-            <a id="enable-authenticator" asp-action="EnableAuthenticator" class="btn btn-primary">@T["Set up authenticator app"]</a>
-            <a id="reset-authenticator" asp-action="ResetAuthenticator" class="btn btn-primary">@T["Reset authenticator app"]</a>
+            @await DisplayAsync(methodShape)
         }
     </div>
-}
-else
-{
-    <div class="alert alert-danger text-center">
-        <strong>@T["Privacy and cookie policy have not been accepted."]</strong>
-        <p>@T["You must accept the policy before you can enable two-factor authentication."]</p>
-    </div>
-}
+
+    @if (Model.IsTwoFaEnabled)
+    {
+        <div class="card-header">
+            @T["Recovery options"]
+        </div>
+
+        <div class="card-body p-0">
+            <div class="row mx-0">
+                <div class="col-auto d-flex align-items-center">
+                    <i class="fa-solid fa-key h1"></i>
+                </div>
+
+                <div class="col p-3">
+
+                    <h4>@T["Recovery codes"]</h4>
+                    <p class="mb-0">@T["Recovery codes can be used to access your account in the event you lose access to your device and cannot receive two-factor verification codes."]</p>
+
+                    @if (Model.RecoveryCodesLeft == 0)
+                    {
+                        <div class="alert alert-danger text-center my-3">
+                            <strong>@T["You have no recovery codes left."]</strong>
+                        </div>
+                    }
+                    else if (Model.RecoveryCodesLeft == 1)
+                    {
+                        <div class="alert alert-danger text-center my-3">
+                            <strong>@T["You have 1 recovery code left."]</strong>
+                        </div>
+                    }
+                    else if (Model.RecoveryCodesLeft <= 3)
+                    {
+                        <div class="alert alert-warning text-center my-3">
+                            <strong>@T["You have {0} recovery {1} left.", Model.RecoveryCodesLeft, T.Plural(Model.RecoveryCodesLeft, "code", "codes")])</strong>
```

Comment:
This should be a single T.Plural with the two full sentences

Reply:
done